pandas/_libs/tslibs/parsing.pyx

```diff
@@ -936,8 +936,7 @@ def guess_datetime_format(dt_str: str, bint dayfirst=False) -> str | None:
         datetime_attrs_to_format.remove(day_attribute_and_format)
         datetime_attrs_to_format.insert(0, day_attribute_and_format)
 
-    # same default used by dateutil
-    default = datetime.now().replace(hour=0, minute=0, second=0, microsecond=0)
+    default = datetime(1970, 1, 1)
```

Comment:
the guessed format is checked explicitly against the input:

https://github.com/pandas-dev/pandas/blob/7bdb6d8498226bc4c55ddde8a6d5ee654c10413f/pandas/_libs/tslibs/parsing.pyx#L1050-L1061

so this should be safe

Reply:
worth adding a comment about why this default was chosen?

Reply:
sure, thanks